homeassistant/components/inkbird/__init__.py

```diff
@@ -2,55 +2,114 @@
 
 from __future__ import annotations
 
+from datetime import datetime, timedelta
 import logging
 
 from inkbird_ble import INKBIRDBluetoothDeviceData, SensorUpdate
 
 from homeassistant.components.bluetooth import (
     BluetoothScanningMode,
     BluetoothServiceInfo,
+    BluetoothServiceInfoBleak,
+    async_ble_device_from_address,
 )
-from homeassistant.components.bluetooth.passive_update_processor import (
-    PassiveBluetoothProcessorCoordinator,
+from homeassistant.components.bluetooth.active_update_processor import (
+    ActiveBluetoothProcessorCoordinator,
 )
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import Platform
-from homeassistant.core import HomeAssistant, callback
+from homeassistant.core import CoreState, HomeAssistant, callback
+from homeassistant.helpers.event import async_track_time_interval
 
 from .const import CONF_DEVICE_TYPE, DOMAIN
 
 PLATFORMS: list[Platform] = [Platform.SENSOR]
 
 _LOGGER = logging.getLogger(__name__)
 
+FALLBACK_POLL_INTERVAL = timedelta(seconds=180)
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up INKBIRD BLE device from a config entry."""
-    address = entry.unique_id
-    assert address is not None
-    device_type: str | None = entry.data.get(CONF_DEVICE_TYPE)
-    data = INKBIRDBluetoothDeviceData(device_type)
+
+class INKBIRDActiveBluetoothProcessorCoordinator(ActiveBluetoothProcessorCoordinator):
```

Comment:
Would be nicer to move this into a coordinator.py and convert to runtime_data ... but for a followup